### PR TITLE
Upon closing prefs window - don't reload packages unless package paths are modified

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

https://jira.autodesk.com/browse/DYN-3802
Modified command implementation for SavePaths so that when prefs window is closed packages are not reloaded unless paths sequence is modified.

test is added.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### CODE REVIEW

**Review by sm6srw — APPROVED** [review]
LGTM!

**Comment by mjkkirschner** [comment]
running serial tests:
https://master-15.jenkins.autodesk.com/view/DYN/job/DYN-DevCI_Self_Service/812/

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs [hunk 1]
```diff
@@ -226,10 +226,15 @@ private void RemovePathAt(int index)
 
         private void CommitChanges(object param)
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs:227]: Is this method called every time the preferences dialog is closed even though the package preferences tab isn't touched?

Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs:227]: yes
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs#L94

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs [hunk 2]
```diff
@@ -226,10 +226,15 @@ private void RemovePathAt(int index)
...
-            if (this.packageLoader != null)
+            var newpaths = CommitRootLocations();
+            //if paths are modified, reload packages and update prefs.
+            if (!setting.CustomPackageFolders.SequenceEqual(newpaths))
             {
-                this.packageLoader.LoadCustomNodesAndPackages(loadPackageParams, customNodeManager);
+                setting.CustomPackageFolders = newpaths;
+                if (this.packageLoader != null)
+                {
+                    this.packageLoader.LoadCustomNodesAndPackages(loadPackageParams, customNodeManager);
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackagePathViewModel.cs:236]: And where was thic call taking place earlier that caused the packages to reload each time?

**Review by pinzart90 — APPROVED** [review]

**Pull request merged**